pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml

```diff
@@ -619,3 +622,96 @@ title: <%= I18n.t('dashboard_landing_title').inspect %>
       });
     }
   }]);
+
+
+  app.controller('SectionAssessmentsController', ['$scope', '$routeParams', '$window', '$q', '$timeout', '$interval', '$sanitize', 'sectionsService', 'studentsService',
+                                             function($scope, $routeParams, $window, $q, $timeout, $interval, $sanitize, sectionsService, studentsService) {
+    $scope.section = sectionsService.get({id: $routeParams.id});
+    $scope.sections = sectionsService.query();
+    $scope.tab = 'assessments';
+
+    $scope.assessments = sectionsService.assessments({id: $routeParams.id});
+    // error handling
+    $scope.genericError = function(result) {
+      $window.alert("An unexpected error occurred, please try again. If this keeps happening, try reloading the page.");
+    };
+    $scope.section.$promise.catch($scope.genericError);
+    $scope.sections.$promise.catch($scope.genericError);
+    $scope.assessments.$promise.catch($scope.genericError);
+
+    // fill in the course dropdown with the section's default course
+    $scope.section.$promise.then(
+      function(section) {
+        $scope.course_id = section.course.id;
+      }
+    );
+
+    // the ng-select in the nav compares by reference not by value, so we can't just set
+    // selectedSection to section, we have to find it in sections.
+    $scope.sections.$promise.then(
+      function( sections ){
+        $scope.selectedSection = $.grep(sections, function(section) { return (section.id == $routeParams.id);})[0];
+      }
+    );
+
+    $scope.assessmentsLoaded = false;
+    $scope.stages = [];
+
+    $scope.course_list = #{DashboardSection::valid_courses.to_json};
+
+    $scope.correctness_list = {
+      free_response: '#{I18n.t('dashboard_correctness_free_response')}',
+      unsubmitted:   '#{I18n.t('dashboard_correctness_unsubmitted')}',
+      correct:       '#{I18n.t('dashboard_correctness_correct')}',
+      incorrect:     '#{I18n.t('dashboard_correctness_incorrect')}'
+    };
+
+    // wait until we have both the students and the student progress
+    $q.all([$scope.section.$promise, $scope.assessments.$promise]).then(function(){
+      $scope.assessmentsLoaded = true;
+      $scope.findStages();
+    });
+
+    $scope.changeScript = function(scriptId) {
+      $scope.assessmentsLoaded = false;
+      $scope.stages = [];
+
+      $scope.assessments = sectionsService.assessments({id: $routeParams.id, script_id: scriptId});
+
+      $scope.levels = [];
+
+      $scope.assessments.$promise.then(function(){
+        $scope.assessmentsLoaded = true;
+        $scope.findStages();
+
+        $.each($scope.assessments, function(index, assessment) {
+
+          assessment.multi_correct_percent = assessment.multi_correct / assessment.multi_count * 100;
```

Comment:
can multi_count ever be 0? if so maybe special case that to generate 0 instead of NaN.